### ISSUE TITLE
cmd/link: Support absolute import paths for main packages with -X

### ISSUE THREAD

**Issue by tsenart** [golang/go#13676]
##### What version of Go are you using (go version)?
`go1.5.1`

##### What operating system and processor architecture are you using?
`darwin/amd64`

##### What did you do?
```console
$ go get -ldflags="-X github.com/tsenart/hello.Version=v0.0.1" github.com/tsenart/hello
$ hello
```

##### What did you expect to see?
```console
v0.0.1
```

##### What did you see instead?
```

```

**Comment by tsenart** [comment]
The context where this would be useful is when compiling many main packages with `go build ./...` and wanting to inject versions at build time for all of them in a single `-ldflags`. We can simply build each command separately but that slows everything down massively due to the duplicate building of interdependencies.

**Comment by minux** [comment]
The import path for main is always main.

**Comment by tsenart** [comment]
@minux: Yes, I figured that out. Would it be a possibility to make it absolute instead?
